---
title: "eQTL_Detector Report"
author: ''
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: default
---

### QC on the sequence data

```{r QTLtools_bamstat, echo=FALSE, fig.height=3.8, fig.width=20, warning=FALSE, fig.cap= "Quality check on sequence data"}
library(readr)
library(kableExtra)

home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")

knitr::kable(Results_bamstat, caption = "QC summary for all  red") %>%
kable_styling(bootstrap_options = "striped", 
              font_size = 8,
              full_width = T)

par(mfrow=c(1,5),
    las = 2)

for(d in colnames(Results_bamstat)){
  barplot(Results_bamstat[,d],
          main = d,
          col = 116546555)
}

```

```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE, fig.cap= " Quality contro of the sequence data, for each sample (on the x-axis) we have the number of samples (green), the trads passing the QC (read) and the exonics reads (black)"}

par (las=2)
barplt1 <- Results_bamstat[,1:3]
barplt1 <- barplt1[order(barplt1$`Number of sequence reads`),]
barplot(as.matrix(t(barplt1)), col =c(1,2,3),xaxt = "n",
        sub = "Samples")
legend("topleft",
c("Number of sequence reads","Number of mapped reads", "Mapped reads annotated"),
fill =c(3,2,1))

```


```{r echo=FALSE, fig.height=4, fig.width=12, message=FALSE, warning=FALSE, paged.print=FALSE, fig.cap= "Assessment of the sequence data in termns of; reads that passed the filter mapping quality (left),mapped reads falling on the annotation specified on the .bed file (center) and annotations covered by at least one sequence on the total number of annotation of the .bed file (rigth)"}

home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")


par(mfrow=c(1,3))

plot(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`,
     ylab = "Number of sequence reads" ,
     xlab = "Number of mapped reads" ,
     col ="red",
     pch = 16)
abline(lm(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`),
        col = "blue",
        lty = 2)

plot(Results_bamstat$`Number of mapped reads`~ Results_bamstat$`Mapped reads annotated`,
     ylab = "Number of mapped reads" ,
     xlab = "Mapped reads annotated" ,
     col ="blue",
     pch = 16)
abline(lm(Results_bamstat$`Number of mapped reads`~ Results_bamstat$`Mapped reads annotated`),
        col = "red",
        lty = 2)

plot(Results_bamstat$`Number of anotations`~ Results_bamstat$`Number of covered annotations by at least one sequence`,
     ylab = "number of annotations covered by at least one sequencing read" ,
     xlab = "total number of annotations" ,
     col ="darkgreen",
     pch = 16)
abline(lm(Results_bamstat$`Number of anotations`~ Results_bamstat$`Number of covered annotations by at least one sequence`),
        col = "darkgreen",
        lty = 2)

```

\pagebreak

### Matching data.

This step is to make sure the good matching between the sequence (phenotype) and the genotyping data.

```{r mbv, echo=FALSE, fig.height=4, fig.width=20, message=FALSE, warning=FALSE, paged.print=FALSE, fig.cap= "For all possible combinatios by sample(sequence/genotype), for each sample the scatter plot is showing the match of hets and homo calls"}
library(readr)
library(tidyverse)
library(gghighlight)
library(readr)


home <- getwd()
Results_mbv <- file.path(home,"Results/mbv")
# We make a list of mbv files for 
list_mbv <- list.files(path = Results_mbv)

setwd(Results_mbv)

par(mfrow=c(1,4))

for (d in list_mbv) {
  mbv  <- read_table2(d)
  #Fields
  x_mbv <- c(mbv$n_het_consistent/mbv$n_het_covered)
  y_mbv <- c(mbv$n_hom_consistent/mbv$n_hom_covered)
  Sample_id <- c(mbv$SampleID)
  ggdata <- data.frame(cbind(x_mbv,y_mbv))
  row.names(ggdata) <- Sample_id
  colnames(ggdata) <- c("Concordance at het","Concordance at hom")
  #Get the ggplot 
  plot(ggdata$`Concordance at hom`,ggdata$`Concordance at het`,
       xlab ="Concordance at hom" ,
       ylab ="Concordance at het" ,
       col ="blue",
       main=d,
       cex.main = 1.2,
       cex.axis = 1,
       pch = "o",
       ylim = c(0,1),
       xlim = c(0,1))
}


```

\pagebreak

### Quantification

```{r quan, echo=FALSE, fig.cap="Boxplot of the experesion in terms of RPKM for all regios", fig.height=16, fig.width=16, message=FALSE, warning=FALSE}

library(readr)
library(kableExtra)

home <- getwd()

Results_quan <- file.path(home,"Bed-Seq")

setwd(Results_quan)

RPKM_all_bed <- read_table2("RPKM_all_sorted.bed.gz")

length_samples <- length(RPKM_all_bed[1,])
length_samples <- length_samples-7

par(mfrow = c(4,1))

boxplot(RPKM_all_bed[,7:length_samples],
        xlab = "Samples",
        main = "Boxplot of RPKM on non-standarized samples")

dens <- apply(RPKM_all_bed[,7:length_samples], 
              2, density)


plot(NA, xlim= c(0,1000), 
     ylim=range(sapply(dens, "[", "y")),
     main = "PDF of RPKM for all samples")
res <- capture.output(mapply(lines, dens, col=1:length(dens)))


boxplot(-log10(RPKM_all_bed[,7:length_samples]),
        xlab = "Samples",
        main = "Boxplot of - log_10 of  RPKM")

dens <- apply(-log10(RPKM_all_bed[,7:length_samples]), 
              2, density,
              na.rm = T)

plot(NA, xlim= c(0,1), ylim= c(0,0.06),
     main = "PDF of -log_10 RPKM for all samples")
res <- capture.output(mapply(lines, dens, col=1:length(dens)))

```

\pagebreak

### PCA on Genotypes


```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE, fig.cap= "Biplot using the PC1 and PC2 for the Genotypes (left), scree plot of all components (midle) and cumulative prediction by using different components "}

library(knitr)
library(readr)

############# Biplot

home <- getwd()
Results_pca<- file.path(home,"Results/pca")
setwd(Results_pca)

Genotypes_pca <- read_table2("Genotypes_pca.txt.pca")
PCA_geno <- Genotypes_pca[,-1]
PCA_geno <- t(PCA_geno)
PCA_geno <- data.frame(PCA_geno)


par(mfrow=c(1,3))

plot(PCA_geno$X2~PCA_geno$X1,
     frame = F,
     col = "red",
     pch = "o",
     xlab = "PC1",
     ylab = "PC2",
     main = "PCA on variants")
abline(v=0,h=0, lty = 2, col = "blue")
text(PCA_geno$X1,PCA_geno$X2, row.names(PCA_geno), cex=0.6, pos=4, col="red") 


############# Prop Var

Genotypes_pca_stat <- read_table2("Genotypes_pca.txt.pca_stats", 
                                  col_names = FALSE)

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

prop_var <- pca_geno_stat[,2]

l = length(prop_var)

prop_var <- prop_var[-l]

plot(prop_var, type = "o",
     frame = F,
     col = "blue",
     xlab = "Principal component")


############# scree plot

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

cumm_comp <- pca_geno_stat[,3]

l = length(cumm_comp )

cumm_comp  <- cumm_comp [-l]

plot(cumm_comp, type = "o",
     frame = F,
     col = "darkgreen",
     xlab = "Principal component")



```

### PCA on gene expression


```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE, fig.cap= "Biplot using the PC1 and PC2 for the gene expression (left), scree plot of all components (midle) and cumulative prediction by using different components"}

library(knitr)
library(readr)

############# Biplot

home <- getwd()
Results_pca<- file.path(home,"Results/pca")
setwd(Results_pca)

Genotypes_pca <- read_table2("pca.Exp.txt.pca")
PCA_geno <- Genotypes_pca[,-1]
PCA_geno <- t(PCA_geno)
PCA_geno <- data.frame(PCA_geno)


par(mfrow=c(1,3))

plot(PCA_geno$X2~PCA_geno$X1,
     frame = F,
     col = "red",
     pch = "o",
     xlab = "PC1",
     ylab = "PC2",
     main = "PCA on gene expression")
abline(v=0,h=0, lty = 2, col = "blue")
text(PCA_geno$X1,PCA_geno$X2, row.names(PCA_geno), cex=0.6, pos=4, col="red") 




############# Prop Var

Genotypes_pca_stat <- read_table2("pca.Exp.txt.pca_stats", 
                                  col_names = FALSE)

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

prop_var <- pca_geno_stat[,2]

l = length(prop_var)

prop_var <- prop_var[-l]

plot(prop_var, type = "o",
     frame = F,
     col = "blue",
     xlab = "Principal component")


############# scree plot

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

cumm_comp <- pca_geno_stat[,3]

l = length(cumm_comp )

cumm_comp  <- cumm_comp [-l]

plot(cumm_comp, type = "o",
     frame = F,
     col = "darkgreen",
     xlab = "Principal component")


```

\pagebreak

```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
library(kableExtra)
home <- getwd()
Results_pca<- file.path(home,"Results/cis_nominal")

cis_nominals <- read_table2("Results/cis_nominal/nominals.txt", 
    col_names = FALSE)

colnames_cis <- c("phenotype ID",
                  "chromosome ID of the phenotype",
                  "start position of the phenotype",
                  "end position of the phenotype",
                  "strand orientation of the phenotype",
                  "total number of variants tested in cis",
                  "distance between the phenotype and the tested variant",
                  "ID",
                  "chromosome ID of the variant",
                  "start position of the variant",
                  "end position of the variant",
                  "nominal P-value",
                  "regression slope",
                  "flag equal to 1 is the variant is the top variant in cis")

colnames(cis_nominals) <- colnames_cis


subset_cis_nominal <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` == "1")



subset_cis <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` ==1)

knitr::kable(subset_cis[1:30,], col.names = c(1:14),
             caption = "List of to 30 cis eQTL mapped using nominal pass, the columns are,
    1. The phenotype ID,
    2. The chromosome ID of the phenotype,
    3. The start position of the phenotype,
    4. The end position of the phenotype,
    5. The strand orientation of the phenotype,
    6. The total number of variants tested in cis,
    7. The distance between the phenotype and the tested variant (accounting for strand orientation),
    8. The ID of the tested variant,
    9. The chromosome ID of the variant,
    10. The start position of the variant,
    11. The end position of the variant,
    12. The nominal P-value of association between the variant and the phenotype,
    13. The corresponding regression slope,
    14. A binary flag equal to 1 is the variant is the top variant in cis,
") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()

##FDR Correction

library(qvalue)

FDR_cis_nomial <- qvalue(cis_nominals$`nominal P-value`,pi0 = 1)

cis_nominals_FDR <- cis_nominals
cis_nominals_FDR$`nominal P-value` <- FDR_cis_nomial[["qvalues"]]
cis_nominals_FDR <- cis_nominals_FDR[order(cis_nominals_FDR$`nominal P-value`),]

```
### Cis Nominal

#### FDR Correction on nominal p-values

```{r echo=FALSE, fig.height=3.5, fig.width=8, message=FALSE, warning=FALSE, fig.cap= "FDR statistics from the qvalue packages"}

plot(FDR_cis_nomial,
     cex.axis = 0.8,
     cex.lab =0.8)

```

```{bash eval=FALSE, include=FALSE}
mv /home/rstudio/Bed-Seq/Genotypes.vcf /home/rstudio/Genotypes.vcf
mv /home/rstudio/Bed-Seq/genotypes.chr22.vcf.gz /home/rstudio/genotypes.chr22.vcf.gz
```



```{r echo=FALSE, fig.height=4, fig.width=18, message=FALSE, warning=FALSE}
par(mfrow = c(1,3))

o <- -log10(sort(cis_nominals$`nominal P-value`, decreasing = FALSE))
e <- -log10(1:length(o)/length(o))
plot(e, o, pch = 19, cex = 1,
     xlab = expression(Expected~~-log[10](italic(p))),
     ylab = expression(Observed~~-log[10](italic(p))),
     xlim = c(0, max(e)), ylim = c(0,max(o)))
lines(e, e, col = "red")
  
hist(-log10(cis_nominals$`nominal P-value`),
     frame = F,
     xlab = "-log10 nominal P-value",
     sub = "",
     main = "Distribution of the nominal p-values",
     col = "blue",
     breaks = 50)

hist(-log10(subset_cis_nominal$`nominal P-value`),
     frame = F,
     xlab = "-log10 nominal P-value",
     sub = "",
     main = "Distribution of the nominal p-values for the cis hits ",
     col = "red",
     breaks = 50)


par(mfrow = c(1,3))

hist(cis_nominals$`distance between the phenotype and the tested variant`,
     breaks = 30,
     xlab = "Distance from the tested variant",
     col = "223263")

plot(cis_nominals$`regression slope`,cis_nominals$`nominal P-value`,
     pch = 16,
     cex = 0.5,
     xlab = "regression slope",
     ylab ="nominal P-value")
abline(v = 0, lty =2, col = "red")

hist(cis_nominals_FDR$`nominal P-value`, 
     xlab = "q-values",
     main = "",
     breaks = 30,
     col = 222)

```


```{r echo=FALSE, message=FALSE, warning=FALSE}

knitr::kable(cis_nominals_FDR[1:30,], col.names = c(1:14),
             caption = "List of to 30 cis eQTLs mapped using nominal pass and the FDR, the columns are,
    1. The phenotype ID,
    2. The chromosome ID of the phenotype,
    3. The start position of the phenotype,
    4. The end position of the phenotype,
    5. The strand orientation of the phenotype,
    6. The total number of variants tested in cis,
    7. The distance between the phenotype and the tested variant (accounting for strand orientation),
    8. The ID of the tested variant,
    9. The chromosome ID of the variant,
    10. The start position of the variant,
    11. The end position of the variant,
    12. q-value,
    13. The corresponding regression slope,
    14. A binary flag equal to 1 is the variant is the top variant in cis,") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()

```


### Expression level forsignificant variants

At this point in the analysis we will use two different sets to plot the effect on genotype on the gene expression, one with a subset on the chromosome 22 of 40 Geuvadis samples, to test how the pipeline is moving from the different steps, and the dummy data provided by QTLtools developers.


```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)


cis_nominal_top <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` ==1)
cis_nominal_top10 <- cis_nominal_top[order(cis_nominal_top$`nominal P-value`),] 
cis_nominal_top10 <- cis_nominal_top10[1:16,]

vcf <- read.vcfR("/home/rstudio/Genotypes.vcf", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_nomial <- vcf[cis_nominal_top10$ID,]
vcf_top10_cis_nomial <- vcf_top10_cis_nomial[!duplicated(vcf_top10_cis_nomial$ID),]
##We import the RPKM_d

rownames(RPKM_all_bed) <- RPKM_all_bed$gene
RPKM_top10_cis_nomial <- RPKM_all_bed[cis_nominal_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_nomial) <- RPKM_top10_cis_nomial$gene 

merge1 <- (merge(cis_nominal_top10, vcf_top10_cis_nomial, by = 'ID'))
merge1 <- merge1[,-(3:21)]

merge2 <- (merge(merge1, RPKM_top10_cis_nomial, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))

merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]

par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}


```


```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)

cis_nominal_top <- subset(cis_nominals_FDR, cis_nominals_FDR$`flag equal to 1 is the variant is the top variant in cis` ==1)
cis_nominal_top10 <- cis_nominal_top[order(cis_nominal_top$`nominal P-value`),] 
cis_nominal_top10 <- cis_nominal_top10[1:16,]

vcf <- read.vcfR("Genotypes.vcf", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_nomial <- vcf[cis_nominal_top10$ID,]
vcf_top10_cis_nomial <- vcf_top10_cis_nomial[!duplicated(vcf_top10_cis_nomial$ID),]
##We import the RPKM_d

rownames(RPKM_all_bed) <- RPKM_all_bed$gene
RPKM_top10_cis_nomial <- RPKM_all_bed[cis_nominal_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_nomial) <- RPKM_top10_cis_nomial$gene 

merge1 <- (merge(cis_nominal_top10, vcf_top10_cis_nomial, by = 'ID'))
merge1 <- merge1[,-(3:21)]

merge2 <- (merge(merge1, RPKM_top10_cis_nomial, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))

merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]

par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}

```

```{bash echo=FALSE}
wget http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz
```

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(readr)
RPKM_all_bed_d <- read_table2("genes.50percent.chr22.bed.gz")
```

```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)

cis_nominal_top <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` ==1)
cis_nominal_top10 <- cis_nominal_top [order(cis_nominal_top$`nominal P-value`),] 
cis_nominal_top10 <- cis_nominal_top10[1:16,]

vcf <- read.vcfR("genotypes.chr22.vcf.gz", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_nomial <- vcf[cis_nominal_top10$ID,]
vcf_top10_cis_nomial <- vcf_top10_cis_nomial[!duplicated(vcf_top10_cis_nomial$ID),]
##We import the RPKM_d

rownames(RPKM_all_bed_d) <- RPKM_all_bed_d$gene
RPKM_top10_cis_nomial <- RPKM_all_bed_d[cis_nominal_top10$`phenotype ID`,]
rownames(RPKM_top10_cis_nomial) <- RPKM_top10_cis_nomial$gene 

merge1 <- (merge(cis_nominal_top10, vcf_top10_cis_nomial, by = 'ID'))
merge1 <- merge1[,-(3:21)]

merge2 <- (merge(merge1, RPKM_top10_cis_nomial, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed_d)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}


```


```{r echo=FALSE, fig.height=4, fig.width=18, message=FALSE, warning=FALSE}

library(vcfR)
library(knitr)
library(readr)

cis_nominal_top <- subset(cis_nominals_FDR, cis_nominals_FDR$`flag equal to 1 is the variant is the top variant in cis` ==1)
cis_nominal_top10 <- cis_nominal_top [order(cis_nominal_top$`nominal P-value`),] 
cis_nominal_top10 <- cis_nominal_top10[1:16,]

vcf <- read.vcfR("genotypes.chr22.vcf.gz", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_nomial <- vcf[cis_nominal_top10$ID,]
vcf_top10_cis_nomial <- vcf_top10_cis_nomial[!duplicated(vcf_top10_cis_nomial$ID),]
##We import the RPKM_d

rownames(RPKM_all_bed_d) <- RPKM_all_bed_d$gene
RPKM_top10_cis_nomial <- RPKM_all_bed_d[cis_nominal_top10$`phenotype ID`,]
rownames(RPKM_top10_cis_nomial) <- RPKM_top10_cis_nomial$gene 

merge1 <- (merge(cis_nominal_top10, vcf_top10_cis_nomial, by = 'ID'))
merge1 <- merge1[,-(3:21)]

merge2 <- (merge(merge1, RPKM_top10_cis_nomial, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed_d)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}



```

\pagebreak

### Cis Permutations Pass


```{r echo=FALSE, fig.height=4.5, fig.width=18, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
home <- getwd()
Results_Cis_Permutations<- file.path(home,"Results/cis_permutation")
setwd(Results_Cis_Permutations)

cis_permutation <- read_table2("permutation.txt", 
    col_names = FALSE)


colnames_cis_permutation <- c("phenotype ID",
                              "chromosome ID of the phenotype",
                              "start position of the phenotype",
                              "end position of the phenotype",
                              "strand orientation of the phenotype",
                              "total number of variants tested in cis",
                              "distance between the phenotype and the tested variant",
                              "ID",
                              "chromosome ID of the top variant",
                              "start position of the top variant",
                              "end position of the top variant",
                              "number of degrees of freedom used to compute the P-values",
                              "Dummy",
                              "first parameter value of the fitted beta distribution",
                              "second parameter value of the fitted beta distribution",
                              "P-value of association between the phenotype and the top variant in cis",
                              "regression slope",
                              "empirircal P-value",
                              "beta distribution P-value")


colnames(cis_permutation) <- colnames_cis_permutation

par(mfrow = c(1,3))

a <- as.matrix((cis_permutation[,19]))
b <- as.matrix((cis_permutation[,18]))

plot(a,b,
     frame = F,
     fill = T,
     xlab = "Direct method",
     ylab = "Beta approximation")
abline(lm(a~b),
       col ="red")


hist(-log10(cis_permutation$`empirircal P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the empirical P-value",
     col = "blue")

hist(-log10(cis_permutation$`beta distribution P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the beta distribution P-value",
     col = "red")


library(qvalue)

d<- read_table2("permutation.txt", 
    col_names = FALSE)

d=d[!is.na(d$X19),]

d$qval=qvalue(d$X19)$qvalue
hist(d$qval, col = "green",
     xlab = "q-value")

cis_permutation_FDR <- d

colnames(cis_permutation_FDR) <- c("phenotype ID",
                                   "chromosome ID of the phenotype",
                                   "start position of the phenotype",
                                   "end position of the phenotype",
                                   "strand orientation of the phenotype",
                                   "total number of variants tested in cis",
                                   "distance between the phenotype and the tested variant",
                                   "ID of the top variant",
                                   "chromosome ID of the top variant",
                                   "start position of the top variant",
                                   "end position of the top variant",
                                   "number of degrees of freedom used to compute the P-values",
                                   "Dummy",
                                   "first parameter value of the fitted beta distribution",
                                   "second parameter value of the fitted beta distribution",
                                   "P-value of association between the phenotype and the top variant in cis",
                                   "regression slope",
                                   "empirircal P-value",
                                   "beta distribution P-value",
                                   "q-value")


```


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}

ww <- cis_permutation[order(cis_permutation$`beta distribution P-value`),]

ww  <- ww[c(1:30),]

knitr::kable(ww,
             col.names = c(1:19),
             caption = "
    1. The phenotype ID
    2. The chromosome ID of the phenotype
    3. The start position of the phenotype
    4. The end position of the phenotype
    5. The strand orientation of the phenotype
    6. The total number of variants tested in cis
    7. The distance between the phenotype and the tested variant (accounting for strand orientation)
    8. The ID of the top variant
    9. The chromosome ID of the top variant
    10. The start position of the top variant
    11. The end position of the top variant
    12. The number of degrees of freedom used to compute the P-values
    13. Dummy
    14. The first parameter value of the fitted beta distribution
    15. The second parameter value of the fitted beta distribution (it also gives the effective number of independent tests in the region)
    16. The nominal P-value of association between the phenotype and the top variant in cis
    17. The corresponding regression slope
    18. The P-value of association adjusted for the number of variants tested in cis given by the direct method (i.e. empirircal P-value)
    19. The P-value of association adjusted for the number of variants tested in cis given by the fitted beta distribution. We strongly recommend to use this adjusted P-value in any downstream analysis" ) %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()


```


```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}


ww <- cis_permutation_FDR[order(cis_permutation_FDR$`q-value`),]
ww  <- ww[c(1:30),]

knitr::kable(ww,
             col.names = c(1:20),
             caption = "
    1. The phenotype ID
    2. The chromosome ID of the phenotype
    3. The start position of the phenotype
    4. The end position of the phenotype
    5. The strand orientation of the phenotype
    6. The total number of variants tested in cis
    7. The distance between the phenotype and the tested variant (accounting for strand orientation)
    8. The ID of the top variant
    9. The chromosome ID of the top variant
    10. The start position of the top variant
    11. The end position of the top variant
    12. The number of degrees of freedom used to compute the P-values
    13. Dummy
    14. The first parameter value of the fitted beta distribution
    15. The second parameter value of the fitted beta distribution (it also gives the effective number of independent tests in the region)
    16. The nominal P-value of association between the phenotype and the top variant in cis
    17. The corresponding regression slope
    18. The P-value of association adjusted for the number of variants tested in cis given by the direct method (i.e. empirircal P-value)
    19. The P-value of association adjusted for the number of variants tested in cis given by the fitted beta distribution. We strongly recommend to use this adjusted P-value in any downstream analysis
    20. q-value") %>%  
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()


```


\pagebreak

```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)


cis_permutation_top10 <- cis_permutation[order(cis_permutation$`beta distribution P-value`),] 
cis_permutation_top10 <- cis_permutation_top10[1:20,]

vcf <- read.vcfR("Genotypes.vcf", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_permutation <- vcf[cis_permutation_top10$ID,]
vcf_top10_cis_permutation <- vcf_top10_cis_permutation[!duplicated(vcf_top10_cis_permutation$ID),]

###
rownames(RPKM_all_bed) <- RPKM_all_bed$gene
RPKM_top10_cis_permutation <- RPKM_all_bed[cis_permutation_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_permutation) <- RPKM_top10_cis_permutation$gene 

merge1 <- (merge(cis_permutation_top10, vcf_top10_cis_permutation, by = 'ID'))

merge1 <- merge1[,-(3:27)]

merge2 <- (merge(merge1, RPKM_top10_cis_permutation, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}


```


```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)
library(qvalue)


q_value_cis_permutation <- qvalue(cis_permutation$`beta distribution P-value`,pi0 = 1)

cis_permutation$`beta distribution P-value` <- q_value_cis_permutation[["qvalues"]]

cis_permutation_top10 <- cis_permutation[order(cis_permutation$`beta distribution P-value`),] 
cis_permutation_top10 <- cis_permutation_top10[1:20,]

vcf <- read.vcfR("Genotypes.vcf", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_permutation <- vcf[cis_permutation_top10$ID,]
vcf_top10_cis_permutation <- vcf_top10_cis_permutation[!duplicated(vcf_top10_cis_permutation$ID),]

###
rownames(RPKM_all_bed) <- RPKM_all_bed$gene
RPKM_top10_cis_permutation <- RPKM_all_bed[cis_permutation_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_permutation) <- RPKM_top10_cis_permutation$gene 

merge1 <- (merge(cis_permutation_top10, vcf_top10_cis_permutation, by = 'ID'))

merge1 <- merge1[,-(3:27)]

merge2 <- (merge(merge1, RPKM_top10_cis_permutation, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "0|1"] <- "1|0"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}


```

```{r fig.height=4, fig.width=18, message=FALSE, warning=FALSE, include=FALSE}

library(vcfR)
library(knitr)
library(readr)


cis_permutation_top10 <- cis_permutation[order(cis_permutation$`beta distribution P-value`),] 
cis_permutation_top10 <- cis_permutation_top10[1:20,]

vcf <- read.vcfR("genotypes.chr22.vcf.gz", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_permutation <- vcf[cis_permutation_top10$ID,]
vcf_top10_cis_permutation <- vcf_top10_cis_permutation[!duplicated(vcf_top10_cis_permutation$ID),]

###
rownames(RPKM_all_bed_d) <- RPKM_all_bed_d$gene
RPKM_top10_cis_permutation <- RPKM_all_bed_d[cis_permutation_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_permutation) <- RPKM_top10_cis_permutation$gene 

merge1 <- (merge(cis_permutation_top10, vcf_top10_cis_permutation, by = 'ID'))

merge1 <- merge1[,-(3:27)]

merge2 <- (merge(merge1, RPKM_top10_cis_permutation, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed_d)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "1|0"] <- "0|1"
  Genotype <- as.matrix(Genotype)
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}

```

\pagebreak

##### Dummy Data without FRD Correction


```{r echo=FALSE, fig.height=4, fig.width=18, message=FALSE, warning=FALSE}

library(vcfR)
library(knitr)
library(readr)
library(qvalue)


q_value_cis_permutation <- qvalue(cis_permutation$`beta distribution P-value`,pi0 = 1)

cis_permutation$`beta distribution P-value` <- q_value_cis_permutation[["qvalues"]]

cis_permutation_top10 <- cis_permutation[order(cis_permutation$`beta distribution P-value`),] 
cis_permutation_top10 <- cis_permutation_top10[1:20,]

vcf <- read.vcfR("genotypes.chr22.vcf.gz", verbose = FALSE )
vcf<- data.frame(cbind(vcf@fix,vcf@gt))
rownames(vcf) <- vcf$ID
vcf_top10_cis_permutation <- vcf[cis_permutation_top10$ID,]
vcf_top10_cis_permutation <- vcf_top10_cis_permutation[!duplicated(vcf_top10_cis_permutation$ID),]

###
rownames(RPKM_all_bed_d) <- RPKM_all_bed_d$gene
RPKM_top10_cis_permutation <- RPKM_all_bed_d[cis_permutation_top10$`phenotype ID`,]

rownames(RPKM_top10_cis_permutation) <- RPKM_top10_cis_permutation$gene 

merge1 <- (merge(cis_permutation_top10, vcf_top10_cis_permutation, by = 'ID'))

merge1 <- merge1[,-(3:27)]

merge2 <- (merge(merge1, RPKM_top10_cis_permutation, 
                 by.x = "phenotype ID" , 
                 by.y ='gene',
                 all.y = F))
merge2_Genes_SNPs <-  c(paste(merge2$`phenotype ID`,merge2$ID))
rownames(merge2) <- merge2_Genes_SNPs
merge2 <- merge2[,-(1:3)]


par(mfrow=c(1,4))

for (a in merge2_Genes_SNPs) {
  sample_number <- (ncol(RPKM_all_bed_d)-6)
  box1 <- t(merge2[a,])
  Genotype <- box1[c(1:sample_number),]
  Genotype <- as.data.frame(Genotype)
  Genotype$Genotype[Genotype$Genotype == "1|0"] <- "0|1"
  l1 <- length(box1)
  l2 <- (l1- sample_number)+1
  Expression_value <- box1[l2:l1,]
  

  boxplot(as.numeric(Expression_value)~as.matrix(Genotype),
          main = a,
          xlab = "Genotype",
          ylab = "Expression Values",
          col = 0)
  stripchart(Expression_value~as.matrix(Genotype), vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue')
}


```

### Trans nominal and permutatios pass

```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
library(qvalue)
home <- getwd()

Results_trans<- file.path(home,"Results/trans")


trans_nominal_best <- read_table2("Results/trans/trans.nominal.best.txt.gz",
                                  col_names = F)
colnames(trans_nominal_best) <- c("Phenotype ID",
                                  "Dummy",
                                  "Nominal P-value")

trans_nominal_bins <- read_table2("Results/trans/trans.nominal.bins.txt.gz",
                                  col_names = F)
colnames(trans_nominal_bins) <- c("Index",
                                  "Dummy1",
                                  "Dummy2",
                                  "Upper boundaries of the P-value bin",
                                  "Lower boundaries of the P-value",
                                  "Number of tests falling in this bin")

trans_nominal_hits <- read_table2("Results/trans/trans.nominal.hits.txt.gz",
                                  col_names = F)
colnames(trans_nominal_hits) <- c(" Phenotype ID",
                                  "Phenotype chrID",
                                  "Phenotype start",
                                  "Variant ID",
                                  "Variant chrID",
                                  "Variant position",
                                  "Nominal P-value of association",
                                  "Dummy",
                                  "Regression slope")



par(mfrow=c(1,3))

trans_nominals_FDR <- qvalue(trans_nominal_hits$`Nominal P-value of association`,pi0 = 1)


hist(as.matrix(trans_nominal_best[,3]),
     breaks = 50,
     col = "blue",
     main = "Distributions of the nominal P-values on top hits per phenotype",
     xlab = "Nominal P-value")

plot(trans_nominal_bins$`Number of tests falling in this bin`~ trans_nominal_bins$Index,
     frame = F,
     col = "red",
     xlab = "Bin index",
     ylab = "Number of tests falling in this bin",
     cex = 3,
     pch = ".")

hist(as.matrix(trans_nominal_hits[,7]),
     breaks = 30,
     col = "green",
     main = "Distributions of the nominal P-values on hits with a P-value above the specified --threshold (=1e-5)",
     xlab = "Nominal P-value")


```

```{r echo=FALSE, fig.height=4, fig.width=8, message=FALSE, warning=FALSE, paged.print=FALSE}

par(mfrow=c(2,4))

plot(trans_nominals_FDR,
     col ="blue")

```


```{r echo=FALSE}

knitr::kable(trans_nominal_hits[1:20,],
             col.names = c(1:9),
             caption = "Hits on trans nominal pass.    
    1. Phenotype ID
    2. Phenotype chrID
    3. Phenotype start
    4. Variant ID
    5. Variant chrID
    6. Variant position
    7. Nominal P-value of association
    8. Dummy here. Field used in approximated mapping in trans
    9. Regression slope
") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()



```

```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
home <- getwd()
Results_trans<- file.path(home,"Results/trans")


trans_permutation_best <- read_table2("Results/trans/trans.permutation.best.txt.gz",
                                  col_names = F)
colnames(trans_permutation_best) <- c("Phenotype ID",
                                  "Dummy",
                                  "Nominal P-value")

trans_permutation_bins <- read_table2("Results/trans/trans.permutation.bins.txt.gz",
                                  col_names = F)
colnames(trans_permutation_bins) <- c("Index",
                                  "Dummy1",
                                  "Dummy2",
                                  "Upper boundaries of the P-value bin",
                                  "Lower boundaries of the P-value",
                                  "Number of tests falling in this bin")

trans_permutation_hits <- read_table2("Results/trans/trans.permutation.hits.txt.gz",
                                  col_names = F)
colnames(trans_permutation_hits) <- c(" Phenotype ID",
                                  "Phenotype chrID",
                                  "Phenotype start",
                                  "Variant ID",
                                  "Variant chrID",
                                  "Variant position",
                                  "Nominal P-value of association",
                                  "Dummy",
                                  "Regression slope")

```

```{r echo=FALSE, message=FALSE, warning=FALSE}
knitr::kable(trans_permutation_hits[1:20,],
             col.names = c(1:9),
             caption = "Hits on trans permutation pass.    
    1. Phenotype ID
    2. Phenotype chrID
    3. Phenotype start
    4. Variant ID
    5. Variant chrID
    6. Variant position
    7. Nominal P-value of association
    8. Dummy here. Field used in approximated mapping in trans
    9. Regression slope") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down") %>% kableExtra::landscape()




```

### fdensity

```{r echo=FALSE, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
home <- getwd()
Results_fdensity<- file.path(home,"Results/fdensity")
setwd(Results_fdensity)

D <-  read_table2("density.TF.around.QTL.txt", 
    col_names = FALSE)

plot((D$X1+D$X2)/2, D$X3, type="l", xlab="Distance to QTLs", ylab="#annotations/kb") 

```

### fenrich

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(knitr)
library(readr)
home <- getwd()
Results_fenrich<- file.path(home,"Results/fenrich")
setwd(Results_fenrich)
D <- read_table2("enrichement.QTL.in.TF.txt", col_names = FALSE)

knitr::kable(D,             col.names = c(1:8),
             caption = "
    1. observed number of QTLs falling within the functional annotations
    2. total number of QTLs
    3. mean expected number of QTLs falling within the functional annotations (across multiple permutations, see below)
    4. standard deviation of the expected number of QTLs falling within the functional annotations (across multiple permutations, see below)
    5-8. dummy fields
") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down")
fisher.test(matrix(c(D$X1, D$X2, round(D$X3), D$X2), ncol=2))

```

### rtc

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(knitr)
library(readr)
home <- getwd()
Results_rtc<- file.path(home,"Results/rtc")
setwd(Results_rtc)
rtc_results <- read_table2("rtc_results.txt")


knitr::kable(rtc_results[1:30,], col.names = c(1:22),caption = "Overlat GWAS hits with eQTL's.
    1. GWAS variant
    2. eQTL variant
    3. Gene
    4. Gene Group
    5. GWAS variant chromosome
    6. GWAS variant position
    7. GWAS variant rank
    8. eQTL variant chromosome
    9. eQTL variant position
    10. eQTL variant rank
    11. Gene chromosome
    12. Gene position
    13. Distance between variants
    14. Distance between GWAS and phenotype
    15. GWAS variant's region index
    16. eQTL variant's region index
    17. Region start
    18. Region end
    19. Number of variants in the region
    20. RTC
    21. D prime
    22. R squared") %>% 
  kable_styling(font_size = 6,
                full_width = F,
                fixed_thead = T,
                latex_options = "scale_down" ) %>% kableExtra::landscape()

```

