---
title: "eQTL_Detector Report"
author: ""
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

## QC on the sequence data

```{r QTLtools_bamstat, echo=FALSE, fig.height=8, fig.width=14, warning=FALSE}
library(readr)
library(kableExtra)

home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")

knitr::kable(Results_bamstat, caption = "QC summary for each red")
par(mfrow=c(2,3),
    las = 2)

for(d in colnames(Results_bamstat)){
  barplot(Results_bamstat[,d],
          main = d,
          col = 116546555)
}


```




```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE, paged.print=FALSE}


home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")


par(mfrow=c(1,3))

plot(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`,
     ylab = "Number of sequence reads" ,
     xlab = "Number of mapped reads" ,
     col ="red")

plot(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`,
     ylab = "Number of sequence reads" ,
     xlab = "Number of mapped reads" ,
     col ="red")

plot(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`,
     ylab = "Number of sequence reads" ,
     xlab = "Number of mapped reads" ,
     col ="red")

```


## Matching data.


```{r mbv, echo=FALSE, fig.width=9, message=FALSE, warning=FALSE, paged.print=TRUE}
library(readr)
library(tidyverse)
library(gghighlight)


home <- getwd()
Results_mbv <- file.path(home,"Results/mbv")
# We make a list of mbv files for 
list_mbv <- list.files(path = Results_mbv)

setwd(Results_mbv)

par(mfrow=c(2,3))

for (d in list_mbv) {
  mbv  <- read_table2(d)
  #Fields
  x_mbv <- c(mbv$n_het_consistent/mbv$n_het_covered)
  y_mbv <- c(mbv$n_hom_consistent/mbv$n_hom_covered)
  Sample_id <- c(mbv$SampleID)
  ggdata <- data.frame(cbind(x_mbv,y_mbv))
  row.names(ggdata) <- Sample_id
  colnames(ggdata) <- c("Concordance at het","Concordance at hom")
  #Get the ggplot 
  plot(ggdata$`Concordance at hom`,ggdata$`Concordance at het`,
       xlab ="Concordance at het" ,
       ylab ="Concordance at hom" ,
       col ="blue",
       main=d,
       cex.lab = 0.8,
       cex.main = 0.7,
       cex.axis = 0.7,
       pch = "x")
}


```


## Quantification

```{r quan, echo=FALSE, fig.height=12, fig.width=16, message=FALSE, warning=FALSE}

library(readr)
library(kableExtra)

home <- getwd()

Results_quan <- file.path(home,"Results/quan")

list_quan <- list.files(path = Results_quan, pattern =".stats")


setwd(Results_quan)

list_quan <- lapply(list_quan, function(x){
  read_delim(x, "\t", escape_double = FALSE, trim_ws = TRUE)
  })

Results_quan_data <- do.call("rbind", lapply(list_quan, as.data.frame))


par(mfrow=c(3,3))


vector_quan <- colnames(Results_quan_data)

for(d in vector_quan[2:10]){
  barplot(Results_quan_data[,d],
          main = d,
          col = "blue")
}



```



## PCA 

```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)

############# Biplot

home <- getwd()
Results_pca<- file.path(home,"Results/pca")
setwd(Results_pca)

Genotypes_pca <- read_table2("Genotypes_pca.txt.pca")
PCA_geno <- Genotypes_pca[,-1]
PCA_geno <- t(PCA_geno)
PCA_geno <- data.frame(PCA_geno)


par(mfrow=c(1,3))

plot(PCA_geno$X2~PCA_geno$X1,
     frame = F,
     col = "red",
     pch = "o",
     xlab = "PC1",
     ylab = "PC2",
     main = "PCA on variants")
abline(v=0,h=0, lty = 2, col = "blue")
text(PCA_geno$X1,PCA_geno$X2, row.names(PCA_geno), cex=0.6, pos=4, col="red") 


############# Prop Var

Genotypes_pca_stat <- read_table2("Genotypes_pca.txt.pca_stats", 
                                  col_names = FALSE)

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

prop_var <- pca_geno_stat[,2]

l = length(prop_var)

prop_var <- prop_var[-l]

plot(prop_var, type = "o",
     frame = F,
     col = "blue",
     xlab = "Principal component")


############# scree plot

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

cumm_comp <- pca_geno_stat[,3]

l = length(cumm_comp )

cumm_comp  <- cumm_comp [-l]

plot(cumm_comp, type = "o",
     frame = F,
     col = "darkgreen",
     xlab = "Principal component")

```

## Cis nominal


```{r}

library(knitr)
library(readr)

home <- getwd()

Results_cis_nominal<- file.path(home,"Results/cis_nominal")

nominals <- read_table2("Results/cis_nominal/nominals.txt", 
    col_names = FALSE)


colnames_cis_nominal <- c("phenotype ID",
                          "chromosome ID",
                          "start position",
                          "end position",
                          "strand orientation",
                          "number of variants in cis",
                          "distance phenotype to variant",
                          "ID of the tested variant",
                          "chromosome ID of the variant",
                          "start position of the variant",
                          "end position of the variant",
                          "nominal P-value",
                          "regression slope",
                          "binary")


colnames(nominals) <- colnames_cis_nominal


plot(density(nominals$`nominal P-value`), frame = F,
     col = "blue",
     main = "PDF of the nominal p-values")

export_nominal.bed <- as.data.frame(cbind(nominals$`chromosome ID`,
                                      nominals$`start position`,
                                      nominals$`end position`,
                                      nominals$`ID of the tested variant`,
                                      nominals$`phenotype ID`,
                                      nominals$`strand orientation`))


write.table( export_nominal.bed, file = "cis_nomina.bed", sep=" ",  col.names=FALSE)



```








