---
title: "eQTL_Detector Report"
author: ''
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  word_document: default
  pdf_document: default
---

## QC on the sequence data


```{r QTLtools_bamstat, echo=FALSE, fig.height=8, fig.width=14, warning=FALSE}

library(readr)
library(kableExtra)

home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")

knitr::kable(Results_bamstat, caption = "QC summary for each red") %>%
kable_styling(bootstrap_options = "striped", 
              font_size = 8,
              full_width = T)

par(mfrow=c(2,3),
    las = 2)

for(d in colnames(Results_bamstat)){
  barplot(Results_bamstat[,d],
          main = d,
          col = 116546555)
}


allPackages <- installed.packages()[, 1]
findBrokenPackages <- function(packages) {
    for (p in packages) {
        tryCatch(ncol(asNamespace(p)$.__NAMESPACE__.$S3methods),
                 error = function(e) print(c(p, e)))
    }
}
findBrokenPackages(allPackages)
```



```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}

par (las=2)
barplt1 <- Results_bamstat[,1:3]
barplt1 <- barplt1[order(barplt1$`Number of sequence reads`),]
barplot(as.matrix(t(barplt1)), col =c(1,2,3),xaxt = "n",
        sub = "Samples")
legend("topleft",
c("Number of sequence reads","Number of mapped reads", "Mapped reads annotated"),
fill =c(3,2,1))

```


```{r echo=FALSE, fig.height=5, fig.width=14, message=FALSE, warning=FALSE, paged.print=FALSE}


home <- getwd()
Results_bamstat <- file.path(home,"Results/bamstat")

list_bamstat <- list.files(path = Results_bamstat)

setwd(Results_bamstat)
list_bamstatII <- lapply(list_bamstat, function(x){
  read.table(file = x, 
             quote="\"", 
             comment.char="")
  })



Results_bamstat <- do.call("rbind", lapply(list_bamstatII, as.data.frame))
rownames(Results_bamstat) <- list_bamstat
colnames(Results_bamstat) <- c("Number of sequence reads",
                               "Number of mapped reads",
                               "Mapped reads annotated",
                               "Number of anotations",
                               "Number of covered annotations by at least one sequence")


par(mfrow=c(1,2))

plot(Results_bamstat$`Number of sequence reads`~Results_bamstat$`Number of mapped reads`,
     ylab = "Number of sequence reads" ,
     xlab = "Number of mapped reads" ,
     col ="red")

plot(Results_bamstat$`Number of mapped reads`~ Results_bamstat$`Mapped reads annotated`,
     ylab = "Number of mapped reads" ,
     xlab = "Mapped reads annotated" ,
     col ="blue")

```


## Matching data.


```{r mbv, echo=FALSE, fig.height=4, fig.width=20, message=FALSE, warning=FALSE, paged.print=FALSE}
library(readr)
library(tidyverse)
library(gghighlight)


home <- getwd()
Results_mbv <- file.path(home,"Results/mbv")
# We make a list of mbv files for 
list_mbv <- list.files(path = Results_mbv)

setwd(Results_mbv)

par(mfrow=c(1,4))

for (d in list_mbv) {
  mbv  <- read_table2(d)
  #Fields
  x_mbv <- c(mbv$n_het_consistent/mbv$n_het_covered)
  y_mbv <- c(mbv$n_hom_consistent/mbv$n_hom_covered)
  Sample_id <- c(mbv$SampleID)
  ggdata <- data.frame(cbind(x_mbv,y_mbv))
  row.names(ggdata) <- Sample_id
  colnames(ggdata) <- c("Concordance at het","Concordance at hom")
  #Get the ggplot 
  plot(ggdata$`Concordance at hom`,ggdata$`Concordance at het`,
       xlab ="Concordance at het" ,
       ylab ="Concordance at hom" ,
       col ="blue",
       main=d,
       cex.main = 1.2,
       cex.axis = 1,
       pch = "o",
       ylim = c(0,1),
       xlim = c(0,1))
}


```

## Quantification

```{r quan, echo=FALSE, fig.height=8, fig.width=16, message=FALSE, warning=FALSE}

library(readr)
library(kableExtra)

home <- getwd()

Results_quan <- file.path(home,"Bed-Seq")

setwd(Results_quan)

RPKM_all_bed <- read_table2("RPKM_all_sorted.bed.gz")

length_samples <- length(RPKM_all_bed[1,])
length_samples <- length_samples-7

par(mfrow = c(2,1))

boxplot(RPKM_all_bed[,7:length_samples],
        xlab = "Samples",
        main = "Boxplot of RPKM on non-standarized samples")

boxplot(-log10(RPKM_all_bed[,7:length_samples]),
        xlab = "Samples",
        main = "Boxplot of - log_10 of  RPKM")

```

## PCA on Genotypes

```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)

############# Biplot

home <- getwd()
Results_pca<- file.path(home,"Results/pca")
setwd(Results_pca)

Genotypes_pca <- read_table2("Genotypes_pca.txt.pca")
PCA_geno <- Genotypes_pca[,-1]
PCA_geno <- t(PCA_geno)
PCA_geno <- data.frame(PCA_geno)


par(mfrow=c(1,3))

plot(PCA_geno$X2~PCA_geno$X1,
     frame = F,
     col = "red",
     pch = "o",
     xlab = "PC1",
     ylab = "PC2",
     main = "PCA on variants")
abline(v=0,h=0, lty = 2, col = "blue")
text(PCA_geno$X1,PCA_geno$X2, row.names(PCA_geno), cex=0.6, pos=4, col="red") 


############# Prop Var

Genotypes_pca_stat <- read_table2("Genotypes_pca.txt.pca_stats", 
                                  col_names = FALSE)

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

prop_var <- pca_geno_stat[,2]

l = length(prop_var)

prop_var <- prop_var[-l]

plot(prop_var, type = "o",
     frame = F,
     col = "blue",
     xlab = "Principal component")


############# scree plot

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

cumm_comp <- pca_geno_stat[,3]

l = length(cumm_comp )

cumm_comp  <- cumm_comp [-l]

plot(cumm_comp, type = "o",
     frame = F,
     col = "darkgreen",
     xlab = "Principal component")



```

## PCA on gene expression





```{r echo=FALSE, fig.height=4, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)

############# Biplot

home <- getwd()
Results_pca<- file.path(home,"Results/pca")
setwd(Results_pca)

Genotypes_pca <- read_table2("pca.Exp.txt.pca")
PCA_geno <- Genotypes_pca[,-1]
PCA_geno <- t(PCA_geno)
PCA_geno <- data.frame(PCA_geno)


par(mfrow=c(1,3))

plot(PCA_geno$X2~PCA_geno$X1,
     frame = F,
     col = "red",
     pch = "o",
     xlab = "PC1",
     ylab = "PC2",
     main = "PCA on gene expression")
abline(v=0,h=0, lty = 2, col = "blue")
text(PCA_geno$X1,PCA_geno$X2, row.names(PCA_geno), cex=0.6, pos=4, col="red") 




############# Prop Var

Genotypes_pca_stat <- read_table2("pca.Exp.txt.pca_stats", 
                                  col_names = FALSE)

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

prop_var <- pca_geno_stat[,2]

l = length(prop_var)

prop_var <- prop_var[-l]

plot(prop_var, type = "o",
     frame = F,
     col = "blue",
     xlab = "Principal component")


############# scree plot

pca_geno_stat <- as.matrix(Genotypes_pca_stat[,-1])

pca_geno_stat <- t(pca_geno_stat)

cumm_comp <- pca_geno_stat[,3]

l = length(cumm_comp )

cumm_comp  <- cumm_comp [-l]

plot(cumm_comp, type = "o",
     frame = F,
     col = "darkgreen",
     xlab = "Principal component")


```

## Cis Nominal

```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
library(kableExtra)
home <- getwd()
Results_pca<- file.path(home,"Results/cis_nominal")

cis_nominals <- read_table2("Results/cis_nominal/nominals.txt", 
    col_names = FALSE)

colnames_cis <- c("phenotype ID",
                  "chromosome ID of the phenotype",
                  "start position of the phenotype",
                  "end position of the phenotype",
                  "strand orientation of the phenotype",
                  "total number of variants tested in cis",
                  "distance between the phenotype and the tested variant",
                  "ID of the tested variant",
                  "chromosome ID of the variant",
                  "start position of the variant",
                  "end position of the variant",
                  "nominal P-value",
                  "regression slope",
                  "flag equal to 1 is the variant is the top variant in cis")

colnames(cis_nominals) <- colnames_cis


subset_cis_nominal <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` == "1")

par(mfrow = c(1,3))

o <- -log10(sort(cis_nominals$`nominal P-value`, decreasing = FALSE))
e <- -log10(1:length(o)/length(o))
plot(e, o, pch = 19, cex = 1,
     xlab = expression(Expected~~-log[10](italic(p))),
     ylab = expression(Observed~~-log[10](italic(p))),
     xlim = c(0, max(e)), ylim = c(0,max(o)))
lines(e, e, col = "red")
  
hist(-log10(cis_nominals$`nominal P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the nominal p-values",
     col = "blue",
     breaks = 50)

hist(-log10(subset_cis_nominal$`nominal P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the nominal p-values for the cis hits ",
     col = "red",
     breaks = 50)


subset_cis <- subset(cis_nominals, cis_nominals$`flag equal to 1 is the variant is the top variant in cis` ==1)

knitr::kable(subset_cis[1:20,]) %>%
kable_styling(bootstrap_options = "striped", 
              font_size = 7,
              full_width = T,
              fixed_thead = T)





```

## Cis Permutations Pass

```{r echo=FALSE, fig.height=4.5, fig.width=18, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
home <- getwd()
Results_Cis_Permutations<- file.path(home,"Results/cis_permutation")
setwd(Results_Cis_Permutations)

cis_permutation <- read_table2("permutation.txt", 
    col_names = FALSE)


colnames_cis_permutation <- c("phenotype ID",
                              "chromosome ID of the phenotype",
                              "start position of the phenotype",
                              "end position of the phenotype",
                              "strand orientation of the phenotype",
                              "total number of variants tested in cis",
                              "distance between the phenotype and the tested variant",
                              "ID of the top variant",
                              "chromosome ID of the top variant",
                              "start position of the top variant",
                              "end position of the top variant",
                              "number of degrees of freedom used to compute the P-values",
                              "Dummy",
                              "first parameter value of the fitted beta distribution",
                              "second parameter value of the fitted beta distribution",
                              "P-value of association between the phenotype and the top variant in cis",
                              "regression slope",
                              "empirircal P-value",
                              "beta distribution P-value")


colnames(cis_permutation) <- colnames_cis_permutation

par(mfrow = c(1,4))

a <- as.matrix((cis_permutation[,19]))
b <- as.matrix((cis_permutation[,18]))

plot(a,b,
     frame = F,
     fill = T)
abline(lm(a~b),
       col ="red")


hist(-log10(cis_permutation$`empirircal P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the empirical P-value",
     col = "blue")

hist(-log10(cis_permutation$`beta distribution P-value`),
     frame = F,
     xlab = "nominal P-value",
     sub = "",
     main = "Distribution of the beta distribution P-value",
     col = "red")




library(qvalue)

d<- read_table2("permutation.txt", 
    col_names = FALSE)

d=d[!is.na(d$X19),]

d$qval=qvalue(d$X19)$qvalue
hist(d$qval,
     col = "green")

write.table(d[which(d$qval <= 0.05), ], "results.genes.significant.txt", quote=FALSE, row.names=FALSE, col.names=FALSE)


```

## Trans nominal

```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}

library(knitr)
library(readr)
home <- getwd()
Results_trans<- file.path(home,"Results/trans")


trans_nominal_best <- read_table2("Results/trans/trans.nominal.best.txt.gz",
                                  col_names = F)
colnames(trans_nominal_best) <- c("Phenotype ID",
                                  "Dummy",
                                  "Nominal P-value")

trans_nominal_bins <- read_table2("Results/trans/trans.nominal.bins.txt.gz",
                                  col_names = F)
colnames(trans_nominal_bins) <- c("Index",
                                  "Dummy1",
                                  "Dummy2",
                                  "Upper boundaries of the P-value bin",
                                  "Lower boundaries of the P-value",
                                  "Number of tests falling in this bin")

trans_nominal_hits <- read_table2("Results/trans/trans.nominal.hits.txt.gz",
                                  col_names = F)
colnames(trans_nominal_hits) <- c(" Phenotype ID",
                                  "Phenotype chrID",
                                  "Phenotype start",
                                  "Variant ID",
                                  "Variant chrID",
                                  "Variant position",
                                  "Nominal P-value of association",
                                  "Dummy",
                                  "Regression slope")

```

## Trans permutation

```{r echo=FALSE, fig.height=5, fig.width=16, message=FALSE, warning=FALSE}
library(knitr)
library(readr)
home <- getwd()
Results_trans<- file.path(home,"Results/trans")


trans_permutation_best <- read_table2("Results/trans/trans.permutation.best.txt.gz",
                                  col_names = F)
colnames(trans_permutation_best) <- c("Phenotype ID",
                                  "Dummy",
                                  "Nominal P-value")

trans_permutation_bins <- read_table2("Results/trans/trans.permutation.bins.txt.gz",
                                  col_names = F)
colnames(trans_permutation_bins) <- c("Index",
                                  "Dummy1",
                                  "Dummy2",
                                  "Upper boundaries of the P-value bin",
                                  "Lower boundaries of the P-value",
                                  "Number of tests falling in this bin")

trans_permutation_hits <- read_table2("Results/trans/trans.permutation.hits.txt.gz",
                                  col_names = F)
colnames(trans_permutation_hits) <- c(" Phenotype ID",
                                  "Phenotype chrID",
                                  "Phenotype start",
                                  "Variant ID",
                                  "Variant chrID",
                                  "Variant position",
                                  "Nominal P-value of association",
                                  "Dummy",
                                  "Regression slope")
```





